crates/sui-core/src/consensus_adapter.rs

```diff
@@ -239,50 +239,30 @@ impl ConsensusAdapter {
         ourselves: &AuthorityName,
         tx_digest: &TransactionDigest,
     ) -> Duration {
-        let (nominator, denominator) =
-            Self::position_submit_certificate(committee, ourselves, tx_digest);
-        const MAX_DELAY_MS: u64 = 60_000;
-        // There is no overflow as long as total_stake < u64::MAX / MAX_DELAY_MS, which seem to be reasonable expectation
-        let duration_ms = MAX_DELAY_MS * nominator / denominator;
-        Duration::from_millis(duration_ms)
+        let position = Self::position_submit_certificate(committee, ourselves, tx_digest);
+        const MAX_DELAY: Duration = Duration::from_secs(70);
+        const DELAY_STEP: Duration = Duration::from_secs(7);
+        let duration = DELAY_STEP * (position as u32/*num validators is less then u32::MAX*/);
```

Comment:
May as well `DELAY_STEP * std::cmp::min(position, MAX_DELAY_MULTIPLE)` to avoid the overflow concern & comment?